asab/metrics/service.py

```diff
@@ -16,27 +16,32 @@
 
 #
 
+
 def metric_dimension(metric_name, tags):
 	dim = metric_name
 	if tags is not None:
 		for k in sorted(tags.keys()):
 			dim += ',{}={}'.format(k, tags[k])
 	return dim
 
-#
 
-class MetricsService(asab.Service):
+class MetricsService(asab.Service, asab.ConfigObject):
 
-	def __init__(self, app, service_name):
+	ConfigDefaults = {
+		'seconds_to_flush': 60,
+	}
+
+	def __init__(self, app, service_name, *, config_section_name="service:MetricsService", config=None):
 		super().__init__(app, service_name)
+		asab.ConfigObject.__init__(self, config_section_name, config)
 		
 		self.Metrics = {} # A key is dimension (combination of metric name and tags)
 		self.Targets = []
 		self.Tags = {
 			'host': platform.node(),
 		}
-
-		app.PubSub.subscribe("Application.tick/10!", self._on_flushing_event)
+		self._seconds_to_flush = int(self.Config.get('seconds_to_flush', 60))
+		app.PubSub.subscribe("Application.tick/{}!".format(self._seconds_to_flush), self._on_flushing_event)
```

Comment:
Nope, that cannot be used that way b/c if you specify an incorrect `seconds_to_flush`, it will silently fail (aka do nothing).

Reply:
@ateska What would you suggest here? One of my idea is to use mapping of all application ticks and assert selected value exists in that mapping.
Thanks.

Reply:
Timer :-) There is one in asab.

Reply:
Or don't allow ppl to reconfigure that ... I still don't find any practical use of that ... I see only problems.